app/src/main/java/com/android/shelfLife/model/foodFacts/OpenFoodFactsRepository.kt

```diff
@@ -2,18 +2,68 @@ package com.android.shelfLife.model.foodFacts
 
 import android.util.Log
 import java.io.IOException
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
 import okhttp3.OkHttpClient
 import okhttp3.Request
 import okhttp3.Response
 import org.json.JSONArray
 import org.json.JSONObject
 
+sealed class SearchStatus {
+  data object Idle : SearchStatus()
+
+  data object Loading : SearchStatus()
+
+  data object Success : SearchStatus()
+
+  data object Failure : SearchStatus()
+}
+
 class OpenFoodFactsRepository(
     private val client: OkHttpClient,
     private val baseUrl: String = "https://world.openfoodfacts.net"
 ) : FoodFactsRepository {
 
-  private val MAX_RESULTS = 7 // Adjust the number of results as needed
+  companion object {
+    private val MAX_RESULTS = 7 // Adjust the number of results as needed
+  }
+
+  private val _searchStatus = MutableStateFlow<SearchStatus>(SearchStatus.Idle)
+  override val searchStatus: StateFlow<SearchStatus> = _searchStatus
+
+  private val _foodFactsSuggestions = MutableStateFlow<List<FoodFacts>>(emptyList())
+  override val foodFactsSuggestions: StateFlow<List<FoodFacts>> = _foodFactsSuggestions
+
+  override fun resetSearchStatus() {
+    _searchStatus.value = SearchStatus.Idle
+  }
+
+  // Modified search function
+  override fun searchByBarcode(barcode: Long) {
```

Comment:
great, more business logic correctly moved to the model instead of the viewmodel